### PR TITLE
tests: Add some tests for Yelp

### PR DESCRIPTION
**Pull request by pwithnall**
These cover basic Yelp functionality, plus one of our key additions (the
gnome-shell search provider).

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T28834

### CODE REVIEW

**Comment by pwithnall** [comment]
Example of a passing test run here: https://openqa.endlessm.com/tests/57022

File: needles/yelp_search_provider_results.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "tags": [
+    "yelp_search_provider_results"
+  ],
+  "area": [
+    {
+      "": "Whole screen, minus shell bottom bar (changes between users; clock changes)",
```
Comment (wjt) [inline needles/yelp_search_provider_results.json]: The screenshot to go with this needle does not include a Yelp search result.

I don't think we should be matching the whole screen – I think we should be matching just the search result.

Comment (wjt) [reply needles/yelp_search_provider_results.json]: "Search Google for ____" is not the Yelp provider.

Comment (pwithnall) [reply needles/yelp_search_provider_results.json]: The perils of coming back to OpenQA after context-switching. I added that screenshot as a placeholder and never got back to it. (I know “Search Google for x” is not a Yelp search result.) Ooops :(

**Comment by pwithnall** [comment]
Fixed the incorrect search provider results needle and added a second commit to improve the desktop matching (unrelated to Yelp but generally useful): https://openqa.endlessm.com/tests/57190#step/yelp/1

File: tests/yelp.pm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Run Yelp; it should show our customised main page.
+    type_very_safely("yelp\n");
+    assert_screen('yelp_main_page', 10);
+
+    send_key_combo('alt', 'f4');
+
+    # Search for a help topic, and results should be shown by a search provider.
+    type_very_safely('add user');
+    assert_screen('yelp_search_provider_results', 10);
```
Comment (wjt) [inline tests/yelp.pm]: Might be nice to match one of the results, click on it, and check it works, but that could be something for the future.

**Pull request merged**